langchain/src/prompts/tests/selectors.test.ts

```diff
@@ -0,0 +1,56 @@
+import { expect, test } from "@jest/globals";
+import * as fs from "fs";
+// import * as path from "path";
+import { FakeEmbeddings } from "../../embeddings/fake.js";
+import { LengthBasedExampleSelector } from "../selectors/LengthBasedExampleSelector.js";
+import { SemanticSimilarityExampleSelector } from "../selectors/SemanticSimilarityExampleSelector.js";
+import { Chroma } from "../../vectorstores/index.js";
+import { RecursiveCharacterTextSplitter } from "../../text_splitter.js";
+import { PromptTemplate } from "../prompt.js";
+
+test("Test using LengthBasedExampleSelector", async () => {
+  const prompt = new PromptTemplate({
+    template: "{foo} {bar}",
+    inputVariables: ["foo"],
+    partialVariables: { bar: "baz" },
+  });
+  const selector = new LengthBasedExampleSelector({
+    examplePrompt: prompt,
+    maxLength: 10,
+  });
+  await selector.addExample({ foo: "one two three" });
+  await selector.addExample({ foo: "four five six" });
+  await selector.addExample({ foo: "seven eight nine" });
+  await selector.addExample({ foo: "ten eleven twelve" });
+  const chosen = await selector.selectExamples({ foo: "hello", bar: "world" });
+  expect(chosen).toStrictEqual([
+    { foo: "one two three" },
+    { foo: "four five six" },
+  ]);
+});
+
+test("Test using SemanticSimilarityExampleSelector", async () => {
+  /* Load in the file we want to do question answering over */
+  console.warn("pwd", process.cwd());
+  const text = fs.readFileSync("../examples/state_of_the_union.txt", "utf8");
+  /* Split the text into chunks */
+  const textSplitter = new RecursiveCharacterTextSplitter({ chunkSize: 1000 });
+  const docs = await textSplitter.createDocuments([text]);
+  const vectorStore = await Chroma.fromDocuments(
+    docs,
+    new FakeEmbeddings(),
+    "state_of_the_union"
+  );
```

Comment:
@nfcampos here are the tests - the first one runs fine, but this one i cant test properly because of this Chroma/DuckDB error: https://github.com/chroma-core/chroma/issues/201 i'm not sure what to do here - see if you can test on your end

Reply:
I think we can try HNSWLib instead of Chroma, more self contained, easier for tests